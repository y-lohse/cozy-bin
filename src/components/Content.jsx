import React from 'react'

const Content = ({ text }) => (
  <div className='content-viewer'>
    <pre>{text}</pre>
  </div>
)

export default Content
