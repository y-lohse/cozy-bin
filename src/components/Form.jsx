import React from 'react'
import { Button, Textarea } from 'cozy-ui/react'

class Form extends React.Component {
  onSubmit = e => {
    e.preventDefault()
    this.props.onSubmit({
      content: e.target.content.value
    })
  };

  render () {
    return (
      <form onSubmit={this.onSubmit} className='submit-form'>
        <label for='content'>Content</label>
        <Textarea name='content' id='content' className='textarea' autofocus />
        <Button>Create bin</Button>
      </form>
    )
  }
}

export default Form
