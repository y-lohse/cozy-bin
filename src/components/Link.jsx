import React from 'react'
import { Button, Input } from 'cozy-ui/react'
import CopyToClipboard from 'react-copy-to-clipboard'

class Link extends React.Component {
  state = {
    copied: false
  };

  render ({ url }) {
    const { copied } = this.state
    const theme = copied ? 'highlight' : null

    return (
      <div>
        <Input type='text' value={url} className={'copy-input'} readonly />
        <CopyToClipboard
          text={url}
          onCopy={() => this.setState({ copied: true })}
        >
          <Button theme={theme}>{copied ? 'copied' : 'copy'}</Button>
        </CopyToClipboard>
      </div>
    )
  }
}

export default Link
