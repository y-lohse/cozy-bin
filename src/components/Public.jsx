import React from 'react'

class Public extends React.Component {
  render () {
    return (
      <div className='app-wrapper c-layout'>
        <main className='app-content'>
          hi!
        </main>
      </div>
    )
  }
}

export default Public
