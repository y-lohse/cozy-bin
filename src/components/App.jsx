/* global cozy */
import React from 'react'
import Form from './Form'
import Link from './Link'

class App extends React.Component {
  state = {
    url: null
  }

  createBin = async data => {
    try {
      const doc = await cozy.client.data.create('io.cozy.bin', data)
      const { _id } = doc
      const sharingData = {
        data: {
          type: 'io.cozy.permissions',
          attributes: {
            source_id: 'io.cozy.apps/cozy-bin',
            permissions: {
              bin: {
                type: 'io.cozy.bin',
                verbs: ['GET'],
                values: [_id]
              }
            }
          }
        }
      }
      const resp = await cozy.client.fetchJSON('POST', '/permissions?codes=all', sharingData)
      const code = resp.attributes.codes.all
      const url = new URL(window.location)

      this.setState({
        url: `${url.origin}/bin?sharecode=${code}`
      })
    } catch (e) {
      console.warn(e)
    }
  }

  render () {
    const { url } = this.state

    return (
      <div className='app-wrapper c-layout'>
        <main className='app-content'>
          <div role="contentinfo">
            { url
              ? <Link url={url} />
              : <Form onSubmit={this.createBin} />
            }
          </div>
        </main>
      </div>
    )
  }
}

export default App
