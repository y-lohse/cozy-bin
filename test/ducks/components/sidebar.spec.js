'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { mockT } from '../../lib/I18n'
import { Sidebar } from '../../../src/components/Sidebar'

Enzyme.configure({ adapter: new Adapter() })

describe('Sidebar component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(<Sidebar t={mockT} />).getElement()
    expect(component).toMatchSnapshot()
  })
})
