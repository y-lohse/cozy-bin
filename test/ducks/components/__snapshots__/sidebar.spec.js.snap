// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should be rendered correctly 1`] = `
<aside
  className="o-sidebar"
>
  <nav>
    <ul
      className="c-nav"
    >
      <li
        className="c-nav-item"
      >
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          className="c-nav-link"
          to="/viewhello1"
        >
          <Icon
            className="c-nav-icon"
            icon="test-file-stub"
          />
          First
        </NavLink>
      </li>
      <li
        className="c-nav-item"
      >
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          className="c-nav-link"
          to="/viewhello2"
        >
          <Icon
            className="c-nav-icon"
            icon="test-file-stub"
          />
          Second
        </NavLink>
      </li>
      <li
        className="c-nav-item"
      >
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          className="c-nav-link"
          to="/viewhello3"
        >
          <Icon
            className="c-nav-icon"
            icon="test-file-stub"
          />
          Third
        </NavLink>
      </li>
    </ul>
  </nav>
</aside>
`;
